components/embedder_traits/resources.rs

```diff
@@ -85,38 +136,10 @@ fn resources_for_tests() -> Box<dyn ResourceReaderMethods + Sync + Send> {
             vec![]
         }
         fn read(&self, file: Resource) -> Vec<u8> {
-            let file = match file {
-                Resource::Preferences => "prefs.json",
-                Resource::BluetoothBlocklist => "gatt_blocklist.txt",
-                Resource::DomainList => "public_domains.txt",
-                Resource::HstsPreloadList => "hsts_preload.json",
-                Resource::BadCertHTML => "badcert.html",
-                Resource::NetErrorHTML => "neterror.html",
-                Resource::UserAgentCSS => "user-agent.css",
-                Resource::ServoCSS => "servo.css",
-                Resource::PresentationalHintsCSS => "presentational-hints.css",
-                Resource::QuirksModeCSS => "quirks-mode.css",
-                Resource::RippyPNG => "rippy.png",
-                Resource::MediaControlsCSS => "media-controls.css",
-                Resource::MediaControlsJS => "media-controls.js",
-                Resource::CrashHTML => "crash.html",
-            };
-            let mut path = env::current_exe().unwrap();
-            path = path.canonicalize().unwrap();
-            while path.pop() {
-                path.push("resources");
-                if path.is_dir() {
-                    break;
-                }
-                path.pop();
-            }
+            let file = filename(file);
```

Comment:
This would just be: 

```suggestion
            let file = resource.filename();
```

Reply:
done in 73c665011e20accdd7c4af0833f4e15086aaa88f